types/webpack/index.d.ts

```diff
@@ -919,7 +919,7 @@ declare namespace webpack {
     }
 
     class WatchIgnorePlugin extends Plugin {
-        constructor(paths: RegExp[]);
+        constructor(paths: string[] | RegExp[]);
```

Comment:
I don't think we can lint for this. Sometimes you need a homogeneous array, sometimes you need a heterogeneous array. People gotta know the difference